apis/python/src/tiledbsoma/soma_tiledb_context.py

```diff
@@ -0,0 +1,40 @@
+import os
+from dataclasses import dataclass
+from typing import Optional, cast
+
+import attrs
+import tiledb
+
+
+def build_default_tiledb_ctx() -> tiledb.Ctx:
+    """
+    Build a TileDB context starting with reasonable defaults, and overriding and updating with user-provided config
+    options.
+    """
+
+    # Note: Defaults must provide positive out-of-the-box UX!
+
+    cfg = {"sm.mem.reader.sparse_global_order.ratio_array_data": 0.3}
+
+    # This is necessary for smaller tile capacities when querying with a smaller memory budget.
+
+    # Temp workaround pending https://app.shortcut.com/tiledb-inc/story/23827
+    region = os.getenv("AWS_DEFAULT_REGION")
+    if region is not None:
+        cfg["vfs.s3.region"] = cast(str, region)  # type: ignore
+
+    return tiledb.Ctx(cfg)
+
+
+@attrs.define(frozen=True, kw_only=True)
+class SomaTileDBContext:
```

Comment:
I want to name this in way that captures it is a SOMA-level context and specific to this TileDB-implementation of SOMA, but without confusing it with `tiledb.Ctx`.  Not sure if this name captures the intent, suggestions welcome.